### PR TITLE
Mk 32 Pistol, Mark 1 Rifle, Wakizashi, and more

### PR DESCRIPTION
**Pull request by TadJohnson00**
## About the PR 

This PR adds some weapons promised as part of the gun rework; it's just a small one since it's my first PR and I'm adding these early so that they can be mapped in sooner.

The Mark 32 "Universal" Pistol is a semi-automatic pistol chambered in .25 caseless, feeding from a twelve-round magazine. It's sold by Nanotrasen as a premium sidearm with superior offensive capabilities, lending itself to the role of a concealed weapon for officials, diplomats, and command staff. Must be mapped in manually.

The Mark 1 Rifle (a spiritual successor to the Garand) is a semi-automatic rifle chambered in .30 light, feeding from an internal five-round magazine that can be loaded from stripper clips or by hand. This reliable and much-loved design also comes in tasteful ceremonial colours for command staff. Both options must be mapped in manually.

The Wakizashi is a short Japanese blade, often worn with a Katana as part of Daisho, literally meaning "big-little". Along with the Wakizashi is a newly-sprited Katana to fit a modern art style, and a ceremonial set of sheaths for the pair of swords. This will be found in the Head of Security's locker.

The Chief Medical Officer is getting a chest rig in tasteful medical-themed colours; now the CMO can carry just about anything, and is not limited to purely medical items in their belt. The CMO chest rig will be found in their locker.

Additionally, the double-barrel shotgun will no longer need to be cycled, allowing for faster takedowns of rowdy bar patrons.

Finally, Syndicate agents will now be able to purchase a box of .30 rifle stripper clips from their uplinks, instead of useless AKM magazines that need to be unloaded manually such that their rounds may be used.

**Media**

mk32
mark1
katana
cmo

- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

**Changelog**

:cl: TJohnson (Diamora Jensens)
- add: Added Mk 32 Universal, Mark 1 Rifle and Ceremonial variant, resprited katana, added CMO chest rig.
- remove: Removed the need to cycle double-barrel shotguns.

### CODE REVIEW

**Comment by TadJohnson00** [comment]
Mind helping me with getting these checks successful? First time PR, not certain how to make sure it all goes through smoothly.

**Comment by Elijahrane** [comment]
for the yaml linter you're referencing a nonexistant prototype somewhere
read the error

**Comment by TadJohnson00** [comment]
Pretty sure I fixed the nonexistent prototype, as for the RSI validation, is there an issue with some sprites not getting used in-game? Last I recall, the captain's sabre has similarly unused sprites.

**Comment by TadJohnson00** [comment]
Scratch that, I see the issue.

**Comment by TadJohnson00** [comment]
Hopefully a greentext this time round

**Comment by TadJohnson00** [comment]
Can you give advice on how to fix any issues that are popping up now? I'll most likely get to it ASAP or after a nap.

**Comment by Elijahrane** [comment]
we have non deterministic test failures sometimes, you should be fine

**Comment by TadJohnson00** [comment]
Yay, green! Now I can sleep :3

**Comment by TadJohnson00** [comment]
So what's the next step here, do I close the PR?

**Comment by Elijahrane** [comment]
nah it'll get reviwed or merged when someone gets around with it

**Comment by TadJohnson00** [comment]
Well. This is currently broken, and I'm too tired today to fix it. Greentext part two soon:tm:.

**Comment by TadJohnson00** [comment]
As it stands, I'm unsure if the only issue to do with this PR regards the crash on assert exception related to all firearms. If you could notify me if or when this ever gets fixed so that I can ensure there are no errors on my end, then I'd be greatly appreciative.

**Comment by Vordenburg** [comment]
The merge conflict was removed when you merged master.

**Comment by TadJohnson00** [comment]
So everything should be fine?

**Comment by Colin-Tel** [comment]
I’m not too sure about the HoS getting a katana/wakizashi in their locker (even with a probability). The CMO getting a medical chest rig is more understandable because it has nothing in it.

**Comment by TadJohnson00** [comment]
> I’m not too sure about the HoS getting a katana/wakizashi in their locker (even with a probability). The CMO getting a medical chest rig is more understandable because it has nothing in it.

I figured it would be good to have it as a 100% spawn for a while, just so it could get seen a bit; the katana and wakizashi are both weaker than the captain's sabre, and would offer a more swanky item for the HoS belt slot. If you deem it unnecessary, I will remove it from the locker, I just figured it would be nice to have is all.

**Comment by Colin-Tel** [comment]
> I figured it would be good to have it as a 100% spawn for a while, just so it could get seen a bit; the katana and wakizashi are both weaker than the captain's sabre, and would offer a more swanky item for the HoS belt slot. If you deem it unnecessary, I will remove it from the locker, I just figured it would be nice to have is all.

One thing you could do in this PR is add the katana/wakizashi belt doohickey to Shōkō’s vault, replacing (or in addition to) the two katanas on the rack. Or you could put it *in* the HoS’s Office on the desk. Angle’s vault could get it too. I’d be totally fine with that. 

You could ask other mappers if they want it on their maps, too. I just know for a fact it would fit on Shōkō because I originally requested the wakizashi specifically so I could put it on Shōkō.

**Comment by TadJohnson00** [comment]
I think I'll just remove it from the lockers instead, and let every mapper put it in manually in that case. Looking forward to seeing it in both of those maps though!

**Comment by TadJohnson00** [comment]
Alright, please tell me anything else I need before I can greentext.

**Comment by Elijahrane** [comment]
for a PR with 65 file changes it's probably best to wait until the huge merge finishes in a few days here

**Comment by TadJohnson00** [comment]
it's me again
please, for my sanity
merge

**Comment by Elijahrane** [comment]
test failure. I believe a prototype got renamed, you can check migrations.yml

**Comment by TadJohnson00** [comment]
Fixed a typo

Merge just a week away

**Comment by Vordenburg** [comment]
Is there a reason why all these new entities are occupying upstream files?

**Comment by TadJohnson00** [comment]
> Is there a reason why all these new entities are occupying upstream files?

Occupying upstream files? Can you elaborate?

**Comment by Vordenburg** [comment]
The new entities are not in `Nyanotrasen/` directories. This is going to be a source of merge conflicts in the future.

**Comment by TadJohnson00** [comment]
Ah shit, I see. I'll fix it now

**Comment by TadJohnson00** [comment]
Okay, I moved a bunch of stuff, but kept everything that was intended to override upstream files in the first place - including the katana sprite and double-barrel mechanics.

**Comment by TadJohnson00** [comment]
I don't foresee any issues with the workflow checks, but that may be because I am foreblind.

File: Resources/Prototypes/Catalog/Fills/Lockers/heads.yml
Comment (Elijahrane) [inline Resources/Prototypes/Catalog/Fills/Lockers/heads.yml:123]: Why would CMO of all people get combat webbing?

File: Resources/Prototypes/Nyanotrasen/Catalog/Fills/Items/belt.yml
Comment (Elijahrane) [inline Resources/Prototypes/Nyanotrasen/Catalog/Fills/Items/belt.yml:8]: Is this supposed to have 2 different swords in it or should these be an orGroup?

File: Resources/Prototypes/Nyanotrasen/Entities/Clothing/Belt/belts.yml
Comment (Elijahrane) [inline Resources/Prototypes/Nyanotrasen/Entities/Clothing/Belt/belts.yml:38]: line spacing is bad here

**Review by Elijahrane — CHANGES_REQUESTED** [review]

File: Resources/Prototypes/Catalog/Fills/Lockers/heads.yml
Comment (TadJohnson00) [reply Resources/Prototypes/Catalog/Fills/Lockers/heads.yml:123]: As written in the description, its intended purpose was to allow the CMO - or whoever they choose to give it to - to carry other items in their belt slot, such as crowbars, handheld crew monitors, or shakers/beakers. The reason why I chose webbing instead of a belt is mostly for the purpose of aesthetics, I could simply resprite it if necessary. Or, if you wish for it to be removed, I can arrange that also.

File: Resources/Prototypes/Nyanotrasen/Catalog/Fills/Items/belt.yml
Comment (TadJohnson00) [reply Resources/Prototypes/Nyanotrasen/Catalog/Fills/Items/belt.yml:8]: It is supposed to have both a katana and a wakizashi.

File: Resources/Prototypes/Nyanotrasen/Entities/Clothing/Belt/belts.yml
Comment (TadJohnson00) [reply Resources/Prototypes/Nyanotrasen/Entities/Clothing/Belt/belts.yml:38]: I see, will fix.

File: Resources/Prototypes/Catalog/Fills/Lockers/heads.yml
Comment (Elijahrane) [reply Resources/Prototypes/Catalog/Fills/Lockers/heads.yml:123]: Reduce storage space from 60 to like 30 and it's fine imo

Comment (TadJohnson00) [reply Resources/Prototypes/Catalog/Fills/Lockers/heads.yml:123]: Then consider it done.

**Comment by TadJohnson00** [comment]
Changes have been made, should be good to go.

File: Resources/Prototypes/Nyanotrasen/Catalog/Fills/Boxes/ammunition.yml
Comment (Vordenburg) [inline Resources/Prototypes/Nyanotrasen/Catalog/Fills/Boxes/ammunition.yml:16]: What is the purpose of `Universal` here? It's capitalized, so I assume it means something different than `universal`.

File: Resources/Prototypes/Nyanotrasen/Entities/Clothing/Belt/belts.yml
Comment (Vordenburg) [inline Resources/Prototypes/Nyanotrasen/Entities/Clothing/Belt/belts.yml:42]: Is there a better name than `sheaths`?

File: Resources/Prototypes/Nyanotrasen/Entities/Objects/Weapons/Guns/Ammunition/Magazine/caseless_rifle.yml
Comment (Vordenburg) [inline Resources/Prototypes/Nyanotrasen/Entities/Objects/Weapons/Guns/Ammunition/Magazine/caseless_rifle.yml]: `universal` here but `Universal` elsewhere?

File: Resources/Prototypes/Nyanotrasen/Entities/Objects/Weapons/Guns/Snipers/snipers.yml
Comment (Vordenburg) [inline Resources/Prototypes/Nyanotrasen/Entities/Objects/Weapons/Guns/Snipers/snipers.yml]: Capitalization inconsistencies with other weapons.

File: Resources/Prototypes/Nyanotrasen/Entities/Objects/Weapons/Guns/Snipers/snipers.yml
Comment (Vordenburg) [inline Resources/Prototypes/Nyanotrasen/Entities/Objects/Weapons/Guns/Snipers/snipers.yml]: Capitalization.

File: Resources/Prototypes/Nyanotrasen/Entities/Objects/Weapons/Melee/sword.yml
Comment (Vordenburg) [inline Resources/Prototypes/Nyanotrasen/Entities/Objects/Weapons/Melee/sword.yml]: It's funny that the 2nd paragraph of Wikipedia describes them so:
`Wakizashi are not necessarily just a smaller version of the katana; they could be forged differently and have a different cross section.[[5]](https://en.wikipedia.org/wiki/Wakizashi#cite_note-5)`
Description could be better.

File: Resources/Prototypes/Nyanotrasen/Entities/Clothing/Belt/belts.yml
Comment (Vordenburg) [inline Resources/Prototypes/Nyanotrasen/Entities/Clothing/Belt/belts.yml:68]: Not a fan of giving military hardware to roles that don't need it.

File: Resources/Prototypes/Nyanotrasen/Entities/Objects/Weapons/Guns/Pistols/pistols.yml
Comment (Vordenburg) [inline Resources/Prototypes/Nyanotrasen/Entities/Objects/Weapons/Guns/Pistols/pistols.yml:2]: Is the `Universal` necessary here? The mk 58 is just `mk 58`.

File: Resources/Prototypes/Nyanotrasen/Entities/Objects/Weapons/Guns/Snipers/snipers.yml
Comment (Vordenburg) [inline Resources/Prototypes/Nyanotrasen/Entities/Objects/Weapons/Guns/Snipers/snipers.yml]: Telling the reader what they feel is generally bad writing. Write to evoke the feeling.

File: Resources/Prototypes/Entities/Objects/Weapons/Guns/Shotguns/shotguns.yml
Comment (Vordenburg) [inline Resources/Prototypes/Entities/Objects/Weapons/Guns/Shotguns/shotguns.yml:131]: To be clear, this means the weapon no longer needs the user to click on it to eject a shell, with this set to true? They can just fire one shot after another?

File: Resources/Prototypes/Nyanotrasen/Catalog/Fills/Boxes/ammunition.yml
Comment (TadJohnson00) [reply Resources/Prototypes/Nyanotrasen/Catalog/Fills/Boxes/ammunition.yml:16]: Originally, the pistol was going to be called "Universal", after the Mk23/USP (universal self-loading pistol), but I later decided to combine the two ideas with the name "mk 32 Universal". I'll ensure that all references to the Universal are capitalised.

File: Resources/Prototypes/Nyanotrasen/Entities/Clothing/Belt/belts.yml
Comment (TadJohnson00) [reply Resources/Prototypes/Nyanotrasen/Entities/Clothing/Belt/belts.yml:42]: If there is, I'd be open to suggestion.

File: Resources/Prototypes/Nyanotrasen/Entities/Objects/Weapons/Guns/Ammunition/Magazine/caseless_rifle.yml
Comment (TadJohnson00) [reply Resources/Prototypes/Nyanotrasen/Entities/Objects/Weapons/Guns/Ammunition/Magazine/caseless_rifle.yml]: Will fix.

File: Resources/Prototypes/Nyanotrasen/Entities/Objects/Weapons/Melee/sword.yml
Comment (TadJohnson00) [reply Resources/Prototypes/Nyanotrasen/Entities/Objects/Weapons/Melee/sword.yml]: I'll write something up.

File: Resources/Prototypes/Nyanotrasen/Entities/Clothing/Belt/belts.yml
Comment (TadJohnson00) [reply Resources/Prototypes/Nyanotrasen/Entities/Clothing/Belt/belts.yml:68]: As before, I used webbing instead of a belt mostly due to aesthetics. I also reduced the capacity of the belt per Rane's request. If you would like me to implement further changes, I can do so.

File: Resources/Prototypes/Nyanotrasen/Entities/Objects/Weapons/Guns/Pistols/pistols.yml
Comment (TadJohnson00) [reply Resources/Prototypes/Nyanotrasen/Entities/Objects/Weapons/Guns/Pistols/pistols.yml:2]: As mentioned before, the mk 32 was intended to reference the Mk23/USP, thus the name "Universal".

File: Resources/Prototypes/Nyanotrasen/Entities/Objects/Weapons/Guns/Snipers/snipers.yml
Comment (TadJohnson00) [reply Resources/Prototypes/Nyanotrasen/Entities/Objects/Weapons/Guns/Snipers/snipers.yml]: I'll write up something a little nicer.

File: Resources/Prototypes/Entities/Objects/Weapons/Guns/Shotguns/shotguns.yml
Comment (TadJohnson00) [reply Resources/Prototypes/Entities/Objects/Weapons/Guns/Shotguns/shotguns.yml:131]: That's correct, the weapon no longer needs to be cycled manually and the user can fire both barrels consecutively.

**Comment by TadJohnson00** [comment]
Just want to make sure that it's still a-okay to add the webbing in its current state and keep the Mk32's name as it stands.

**Comment by Elijahrane** [comment]
the name is fine. webbing is w/e, it's not hard to move somewhere else later during the next pass on what drip each head gets

**Review by Elijahrane — APPROVED** [review]

**Pull request merged**
